generator/src/test/java/com/linkedin/pegasus/generator/TestPegasusDataTemplateGenerator.java

```diff
@@ -205,4 +208,123 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
             file -> file.getName().replace(".java", ""),
             Function.identity()));
   }
+
+  @DataProvider(name = "with_typeref_field_in_record_pdsc")
+  private Object[][] createDataTemplateCases()
+  {
+    return new Object[][]
+        {
+            {"IsoDuration.pdsc", "PremiumService.pdsc", "IsoDuration.java", "PremiumService.java", "PremiumServiceState.java", "AccountBalanceType.java"},
+            {"PremiumService.pdsc", "IsoDuration.pdsc", "IsoDuration.java", "PremiumService.java", "PremiumServiceState.java", "AccountBalanceType.java"},
+        };
+  }
+
+  @Test(dataProvider = "with_typeref_field_in_record_pdsc")
+  public void testDataTemplateGenerationOrderWithResolver(String[] testArgs)
+      throws Exception
+  {
+    String[] pegasusFilenames = Arrays.copyOfRange(testArgs, 0, 2);
+    String[] javaFilenames = Arrays.copyOfRange(testArgs, 2, testArgs.length);
+    testDataTemplateGenerationDeterministic(pegasusFilenames, javaFilenames);
+  }
+
+  @DataProvider(name = "with_typeref_field_in_record_pdl")
+  private Object[][] createCasesWithTyperefFieldInRecord()
+  {
+    return new Object[][]
+        {
+            {"AField.pdl", "ARecord.pdl", "AField.java", "ARecord.java"},
+            {"ARecord.pdl", "AField.pdl", "AField.java", "ARecord.java"},
+        };
+  }
+
+  @Test(dataProvider = "with_typeref_field_in_record_pdl")
+  public void testCasesWithTyperefFieldInRecord(String[] testArgs)
+      throws Exception
+  {
+    String[] pegasusFilenames = Arrays.copyOfRange(testArgs, 0, 2);
+    String[] javaFilenames = Arrays.copyOfRange(testArgs, 2, testArgs.length);
+    testDataTemplateGenerationDeterministic(pegasusFilenames, javaFilenames);
+  }
+
+  @DataProvider(name = "with_inline_field_in_record_pdl")
+  private Object[][] createCasesWithInlineFieldInRecord()
+  {
+    return new Object[][]
+        {
+            {"BRecord.pdl", "InlineRecord.java", "BRecord.java"},
+        };
+  }
+
+  @Test(dataProvider = "with_inline_field_in_record_pdl")
+  public void testCasesWithInlineFieldInRecord(String[] testArgs)
+      throws Exception
+  {
+    String[] pegasusFilenames = Arrays.copyOfRange(testArgs, 0, 1);
+    String[] javaFilenames = Arrays.copyOfRange(testArgs, 1, testArgs.length);
+    testDataTemplateGenerationDeterministic(pegasusFilenames, javaFilenames);
+  }
+
+  private void testDataTemplateGenerationDeterministic(String[] pegasusFilenames, String[] javaFilenames)
```

Comment:
A better way to test this (and avoid having to update java files everytime code generation logic changes) is to take two arrays of Pegasus schemas as input (in different order) and then generate the source files and compare they are same.

Reply:
That was my first attempt, sometime today in my chat with Astro, I changed to current way.
I think ...... I can do the original way

Reply:
I think you can just generate MD5 for all the generated files together by xor-ing individual MD5s and compare. It would be simpler than current logic.
See https://stackoverflow.com/questions/47253864/order-independent-hash-in-java

Reply:
I was talking about this comment. What did i suggest?